[FLINK-18979][docs] Update docs to better emphasize remote functions (#132)
apache/flink-statefun: docs/index.md

under the License.
-->

> **Stateful Functions** is a framework, developed under the umbrella of Apache Flink, that makes it simple to build **consistent stateful serverless applications**.

Author: ```suggestion
**Stateful Functions** is a framework, developed under the umbrella of Apache Flink, that simplifies building distributed stateful applications.
```

Other possible first sentence

Reviewer: I personally like the first version more, and have been using that tag line a lot in recent talks / current work-in-progress blog posts. However, I'm also not sure if the first version is "down-scoping" the project as well.

I'd leave this to @igalshilman / @StephanEwen though, as it seems like more a question of how we want the project to be pitched moving forward.

Reviewer: Just my two more cents:
From how I read the changes in the PR, it seems like the overall goal is to make remote functions / the "Flink as DB" mode more prominent and the main focus of the project.

In that sense, I would vote for `**... makes it simple to build consistent stateful serverless applications` as that seems to be more coherent with the goal.

> It is designed to work with modern architectures, like stateless Kubernetes deployments or popular event-driven FaaS platforms like AWS Lambda, KNative, etc.,

Author: ```suggestion
It is designed to work with modern architectures, like cloud-native deployments or popular event-driven FaaS platforms like AWS Lambda, KNative, etc.,
```

This was my other idea for this sentence, I'd like to hear what people like more

Reviewer: Stateless k8s deployments, IMO, would probably catch people a bit confused from a first read.
Overall, I always had the feeling that due to how the project is named "Stateful Functions", every time we use the term "stateless" we are in a bit of risk of misguiding readers who don't fully understand the context the term is used in.

to provide out-of-the-box consistent state and messaging to preserve the serverless experience and elasticity of these platforms. 

<p class="text-center">
    <img width="80%" src="{{ site.baseurl }}/fig/concepts/arch_overview.svg"/>
</p>

It brings together the benefits of Apache Flink®, streaming semantics for processing large datasets with low latency and bounded resource constraints,

> along with a runtime for modeling stateful entities that supports location transparency, concurrency, scaling, and resiliency. 

Author: -1 I think its "resiliency"
